### ISSUE TITLE
Project metadata with invalid pyarrow version pin

### ISSUE THREAD

**Issue by noklam** [googleapis/python-bigquery#1818]
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

Please run down the following list and make sure you've tried the usual "quick fixes":

  - Search the issues already opened: https://github.com/googleapis/python-bigquery/issues
  - Search StackOverflow: https://stackoverflow.com/questions/tagged/google-cloud-platform+python

If you are still having issues, please be sure to include as much information as possible:

#### Environment details

  - OS type and version:
  - Python version: `python --version`
  - pip version: `pip --version`
  - `google-cloud-bigquery` version: `pip show google-cloud-bigquery`

#### Steps to reproduce

  1. `pip install uv`
  2. `uv venv` 
  3. `source .venv/bin/activate`
  4.  `uv pip install kedro-datasets[all]`
  

I am trying out the new package manager `uv`, and from the error it seems to complain about the pin in this package is no a valid version. 

#### Code example

```python
# example
```

#### Stack trace
```
> (.venv) gitpod /workspace/kedro-plugins (main) $ uv pip install -e "kedro-datasets[all]"
>    Built file:///workspace/kedro-plugins/kedro-datasets                                                                                             Built 1 editable in 1.54s
> error: Failed to download: google-cloud-bigquery==1.28.2
>   Caused by: Couldn't parse metadata of google_cloud_bigquery-1.28.2-py2.py3-none-any.whl from https://files.pythonhosted.org/packages/ce/af/89ccb3dd70a86516cb408dd7b7484d2fdd073bdce6405f722f75e6058e66/google_cloud_bigquery-1.28.2-py2.py3-none-any.whl
>   Caused by: after parsing 2.0, found "de" after it, which is not part of a valid version
> pyarrow (<2.0de,>=1.0.0) ; (python_version >= "3.5") and extra == 'all'

```

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

**Comment by astrojuanlu** [comment]
https://github.com/googleapis/python-bigquery/blob/929276949244f704688837dfca35cb7dacd44623/setup.py#L52-L54

See #1402, #1403

Maybe you could just yank that release?

**Comment by Linchin** [comment]
Thanks @astrojuanlu for the investigation! Indeed it seems google-cloud-bigquery v1.28.2 has this typo in the pyarrow version, but it has been fixed in v1.28.3. 

To @noklam, thank you for raising the issue. Since kedro-datasets doesn't pin bigquery to v1.28.2, maybe you can just first install bigquery v1.28.3 manually, and install kedro-datasets? Also, the latest version is v3.17.2, it's probably worth trying to use the later versions of python and bigquery client. Please let us know if you have any further questions :)

**Comment by chalmerlowe** [comment]
I +1 the comment made by @Linchin 
Give v1.28.3 a try OR better yet, move to one of our much more recent versions of python-bigquery. We only rarely make updates to the v1 branch if there is some form of critical fix. It does not get feature additions, etc.

Closing this issue as a work-around/solution has been identified.
